sally/models/stock_production_lot.py

```diff
@@ -0,0 +1,33 @@
+from odoo import models,fields,api
+from odoo.exceptions import ValidationError, UserError
+from datetime import date
+
+class StockProductionLot(models.Model):
+    _inherit = 'stock.lot'
+    water_ids = fields.One2many('flower.water', 'lot_id')
+    is_flower = fields.Boolean("Is Flower", related='product_id.is_flower')
+
+    def action_water_flower(self):
+        flowers = self.browse([])
+        for record in self: 
+            if record.is_flower:
+                flowers += record   
```

Comment:
```suggestion
        flowers = self.filtered(lambda rec: rec.is_flower)
```
Always advised to use the ORM functions as we believe they are more optimized than linear loops that we normally make and also better readability.